lightclient-circuits/src/sync_step_circuit.rs

```diff
@@ -357,37 +337,46 @@ impl<S: Spec, F: Field> StepCircuit<S, F> {
             let participation_bit = ctx.load_witness(F::from(is_attested as u64));
             gate.assert_bit(ctx, participation_bit);
 
-            let assigned_pk = g1_chip.assign_point_unchecked(ctx, pk);
-
-            // *Note:* normally, we would need to take into account the sign of the y coordinate, but
-            // because we are concerned only with signature forgery, if this is the wrong
-            // sign, the signature will be invalid anyway and thus verification fails.
-            /*
-            // Square y coordinate
-            let ysq = fp_chip.mul(ctx, assigned_pk.y.clone(), assigned_pk.y.clone());
-            // Calculate y^2 using the elliptic curve equation
-            let ysq_calc = calculate_ysquared::<F>(ctx, fp_chip, assigned_pk.x.clone());
-            // Constrain witness y^2 to be equal to calculated y^2
-            fp_chip.assert_equal(ctx, ysq, ysq_calc);
-            */
-
-            assigned_affines.push(assigned_pk);
+            let assigned_affine = g1_chip.assign_point(ctx, pk);
+
+            let half_p = fp_chip.load_constant_uint(
+                ctx,
+                modulus::<halo2curves::bls12_381::Fq>() / BigUint::from(2u64),
```

Comment:
Do we need to take into account p is even or odd?

Reply:
no, it's constant for BLS12-381